mqtt/mqtt-broker/tests/common.rs

```diff
@@ -41,12 +51,37 @@ impl TestClient {
         self.subscription_handle.subscribe(subscribe_to).await
     }
 
+    /// Initiates sending Disconnect packet and proper client shutdown.
     pub async fn shutdown(&mut self) -> Result<(), ShutdownError> {
         self.shutdown_handle.shutdown().await
     }
 
+    pub fn shutdown_handle(&mut self) -> ShutdownHandle {
+        self.shutdown_handle.clone()
+    }
+
+    /// Terminates client w/o sending Disconnect packet.
+    pub async fn terminate(self) -> Result<(), JoinError> {
+        self.termination_handle
+            .send(())
+            .expect("unable to send termination signal");
+        self.event_loop_handle.await
+    }
+
+    /// Watis until client's event loop is finished.
     pub async fn join(self) -> Result<(), JoinError> {
-        self.task.await
+        self.event_loop_handle.await
+    }
+}
+
+impl tokio::stream::Stream for TestClient
+where
+    Self: Unpin,
+{
+    type Item = Event;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        self.events_receiver.poll_recv(cx)
```

Comment:
```suggestion
        self.events_receiver.poll_next(cx)
```

Reply:
There is no `.poll_next()` on `UnboundedReceiver`, I think it should be `poll_recv()`